File: 2Sum/Leetcode_1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> d;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (d.find(target - nums[i]) != d.end())
            {
                return {d[target - nums[i]], i};
            }
        d[nums[i]] = i;    
        }
        return {};
    }
};

File: 2Sum/Leetcode_1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        import collections
        d = collections.defaultdict(list)
        for i, num in enumerate(nums):
            d[num].append(i)
        
        for num in nums:
            print(num)
            if target - num == num and len(d[num])>=2: 
                return d[num][:2]
            elif target - num != num and d[target-num]:
                return [d[num][-1], d[target-num][-1]]
